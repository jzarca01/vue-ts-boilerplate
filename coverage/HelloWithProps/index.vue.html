<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for HelloWithProps/index.vue</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">HelloWithProps</a> index.vue
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">91.67% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>22/24</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">38.89% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>14/36</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">75% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>6/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">94.44% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>17/18</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48</td><td class="line-coverage quiet"><span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">&lt;template&gt;<span class="branch-1 cbranch-no" title="branch not covered" ></span>
  &lt;div class="hello"&gt;
    &lt;h1&gt;{<span class="branch-1 cbranch-no" title="branch not covered" >{ message }}&lt;/h1&gt;<span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ><span class="branch-2 cbranch-no" title="branch not covered" ></span></span></span></span>
    <span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ><span class="branch-3 cbranch-no" title="branch not covered" ></span></span></span>
  &lt;/div&gt;
&lt;/template&gt;
&nbsp;
&lt;script lang="ts"&gt;<span class="branch-0 cbranch-no" title="branch not covered" ></span>
import Vue from 'vue';
import { Component } from 'vue-property-decorator';
<span class="branch-1 cbranch-no" title="branch not covered" ></span>
@Component({<span class="branch-1 cbranch-no" title="branch not covered" ></span>
  pr<span class="missing-if-branch" title="else path not taken" >E</span>ops: {
    message: String
  }<span class="branch-2 cbranch-no" title="branch not covered" ><span class="branch-1 cbranch-no" title="branch not covered" ></span></span>
})
&nbsp;
export default class HelloWithProps extends Vue {
&nbsp;
}
&nbsp;
&lt;/script&gt;
&nbsp;
&lt;!-- Add "scoped" attribute to limit CSS to this component only --&gt;
&lt;style scoped&gt;
h1, h2 {
  font-weight: normal;
}
&nbsp;
ul { padding: 0; }
li { list-style-type: none; }
.code { color: #6f92ba; }
&nbsp;
ul.links li {
  list-style-type: none;
  display: inline-block;
  margin: 0 16px;
}
ul.tips li {
  margin-bottom: 10px;
}
&nbsp;
a {
  color: #42b983;
}
&nbsp;
&lt;/style&gt;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Feb 01 2018 17:03:55 GMT+0100 (CET)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
