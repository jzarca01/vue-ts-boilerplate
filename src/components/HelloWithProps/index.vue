<template>
  <div class="hello">
    <h1>{{ message }}</h1>
    
  </div>
</template>

<script lang="ts">
import Vue from 'vue';
import { Component } from 'vue-property-decorator';

@Component({
  props: {
    message: String
  }
})

export default class HelloWithProps extends Vue {

}

</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
h1, h2 {
  font-weight: normal;
}

ul { padding: 0; }
li { list-style-type: none; }
.code { color: #6f92ba; }

ul.links li {
  list-style-type: none;
  display: inline-block;
  margin: 0 16px;
}
ul.tips li {
  margin-bottom: 10px;
}

a {
  color: #42b983;
}

</style>
