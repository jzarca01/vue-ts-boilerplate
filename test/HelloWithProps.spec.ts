import { mount } from "vue-test-utils";
import { shallow } from "@vue/test-utils";
import HelloWithProps from "../src/components/HelloWithProps";

describe("HelloWithProps.spec.js", () => {
    let wrapper: any;

    beforeEach(() => {
        wrapper = mount(HelloWithProps, {
            propsData: {
                message: "Hello World"
            }
        });
    });

    it("renders without crashing", () => {
        const wrapper = shallow(HelloWithProps);
        expect(wrapper.contains('div')).toBe(true);
        expect(wrapper).toMatchSnapshot();
    });

    it("should display Hello World", () => {
        expect(wrapper.vm.message).toEqual("Hello World");
    });
});