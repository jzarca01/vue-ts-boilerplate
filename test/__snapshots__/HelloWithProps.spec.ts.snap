// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWithProps.spec.js renders without crashing 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "element": <div
    class="hello"
  >
    <h1>
      
    </h1>
  </div>,
  "isVueComponent": true,
  "options": Object {
    "attachedToDocument": false,
  },
  "update": [Function],
  "version": 2.5,
  "vm": VueComponent {
    "$_mountingOptionsSlots": undefined,
    "$_originalSlots": Object {},
    "$attrs": Object {},
    "$children": Array [],
    "$createElement": [Function],
    "$el": <div
      class="hello"
    >
      <h1>
        
      </h1>
    </div>,
    "$emit": [Function],
    "$listeners": Object {},
    "$options": Object {
      "_Ctor": Object {
        "0": [Function],
      },
      "_base": [Function],
      "_parentElm": null,
      "_propKeys": Array [
        "message",
      ],
      "_refElm": null,
      "beforeCreate": Array [
        [Function],
      ],
      "components": Object {
        "transition": Object {
          "render": [Function],
        },
        "transition-group": Object {
          "render": [Function],
        },
      },
      "data": [Function],
      "directives": Object {},
      "filters": Object {},
      "mixins": Array [
        Object {
          "data": [Function],
        },
      ],
      "name": "HelloWithProps",
      "props": Object {
        "message": Object {
          "type": [Function],
        },
      },
      "render": [Function],
      "staticRenderFns": Array [],
    },
    "$parent": undefined,
    "$refs": Object {},
    "$root": [Circular],
    "$scopedSlots": Object {},
    "$slots": Object {},
    "$vnode": undefined,
    "__emitted": Object {},
    "__emittedByOrder": Array [],
    "_c": [Function],
    "_data": Object {},
    "_directInactive": false,
    "_events": Object {},
    "_hasHookEvent": false,
    "_inactive": null,
    "_isBeingDestroyed": false,
    "_isDestroyed": false,
    "_isMounted": true,
    "_isVue": true,
    "_props": Object {
      "message": undefined,
    },
    "_renderProxy": VueComponent {
      "$_mountingOptionsSlots": undefined,
      "$_originalSlots": Object {},
      "$attrs": Object {},
      "$children": Array [],
      "$createElement": [Function],
      "$el": <div
        class="hello"
      >
        <h1>
          
        </h1>
      </div>,
      "$emit": [Function],
      "$listeners": Object {},
      "$options": Object {
        "_Ctor": Object {
          "0": [Function],
        },
        "_base": [Function],
        "_parentElm": null,
        "_propKeys": Array [
          "message",
        ],
        "_refElm": null,
        "beforeCreate": Array [
          [Function],
        ],
        "components": Object {
          "transition": Object {
            "render": [Function],
          },
          "transition-group": Object {
            "render": [Function],
          },
        },
        "data": [Function],
        "directives": Object {},
        "filters": Object {},
        "mixins": Array [
          Object {
            "data": [Function],
          },
        ],
        "name": "HelloWithProps",
        "props": Object {
          "message": Object {
            "type": [Function],
          },
        },
        "render": [Function],
        "staticRenderFns": Array [],
      },
      "$parent": undefined,
      "$refs": Object {},
      "$root": [Circular],
      "$scopedSlots": Object {},
      "$slots": Object {},
      "$vnode": undefined,
      "__emitted": Object {},
      "__emittedByOrder": Array [],
      "_c": [Function],
      "_data": Object {},
      "_directInactive": false,
      "_events": Object {},
      "_hasHookEvent": false,
      "_inactive": null,
      "_isBeingDestroyed": false,
      "_isDestroyed": false,
      "_isMounted": true,
      "_isVue": true,
      "_props": Object {
        "message": undefined,
      },
      "_renderProxy": [Circular],
      "_self": [Circular],
      "_staticTrees": null,
      "_uid": 1,
      "_vnode": VNode {
        "asyncFactory": undefined,
        "asyncMeta": undefined,
        "children": Array [
          VNode {
            "asyncFactory": undefined,
            "asyncMeta": undefined,
            "children": Array [
              VNode {
                "asyncFactory": undefined,
                "asyncMeta": undefined,
                "children": undefined,
                "componentInstance": undefined,
                "componentOptions": undefined,
                "context": undefined,
                "data": undefined,
                "elm": ,
                "fnContext": undefined,
                "fnOptions": undefined,
                "fnScopeId": undefined,
                "isAsyncPlaceholder": false,
                "isCloned": false,
                "isComment": false,
                "isOnce": false,
                "isRootInsert": false,
                "isStatic": false,
                "key": undefined,
                "ns": undefined,
                "parent": undefined,
                "raw": false,
                "tag": undefined,
                "text": "",
              },
            ],
            "componentInstance": undefined,
            "componentOptions": undefined,
            "context": [Circular],
            "data": undefined,
            "elm": <h1>
              
            </h1>,
            "fnContext": undefined,
            "fnOptions": undefined,
            "fnScopeId": undefined,
            "isAsyncPlaceholder": false,
            "isCloned": false,
            "isComment": false,
            "isOnce": false,
            "isRootInsert": false,
            "isStatic": false,
            "key": undefined,
            "ns": undefined,
            "parent": undefined,
            "raw": false,
            "tag": "h1",
            "text": undefined,
          },
        ],
        "componentInstance": undefined,
        "componentOptions": undefined,
        "context": [Circular],
        "data": Object {
          "staticClass": "hello",
        },
        "elm": <div
          class="hello"
        >
          <h1>
            
          </h1>
        </div>,
        "fnContext": undefined,
        "fnOptions": undefined,
        "fnScopeId": undefined,
        "isAsyncPlaceholder": false,
        "isCloned": false,
        "isComment": false,
        "isOnce": false,
        "isRootInsert": true,
        "isStatic": false,
        "key": undefined,
        "ns": undefined,
        "parent": undefined,
        "raw": false,
        "tag": "div",
        "text": undefined,
      },
      "_watcher": Watcher {
        "active": true,
        "cb": [Function],
        "deep": false,
        "depIds": Set {
          6,
        },
        "deps": Array [
          Dep {
            "id": 6,
            "subs": Array [
              [Circular],
            ],
          },
        ],
        "dirty": false,
        "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
        "getter": [Function],
        "id": 2,
        "lazy": false,
        "newDepIds": Set {},
        "newDeps": Array [],
        "sync": false,
        "user": false,
        "value": undefined,
        "vm": [Circular],
      },
      "_watchers": Array [
        Watcher {
          "active": true,
          "cb": [Function],
          "deep": false,
          "depIds": Set {
            6,
          },
          "deps": Array [
            Dep {
              "id": 6,
              "subs": Array [
                [Circular],
              ],
            },
          ],
          "dirty": false,
          "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
          "getter": [Function],
          "id": 2,
          "lazy": false,
          "newDepIds": Set {},
          "newDeps": Array [],
          "sync": false,
          "user": false,
          "value": undefined,
          "vm": [Circular],
        },
      ],
    },
    "_self": [Circular],
    "_staticTrees": null,
    "_uid": 1,
    "_vnode": VNode {
      "asyncFactory": undefined,
      "asyncMeta": undefined,
      "children": Array [
        VNode {
          "asyncFactory": undefined,
          "asyncMeta": undefined,
          "children": Array [
            VNode {
              "asyncFactory": undefined,
              "asyncMeta": undefined,
              "children": undefined,
              "componentInstance": undefined,
              "componentOptions": undefined,
              "context": undefined,
              "data": undefined,
              "elm": ,
              "fnContext": undefined,
              "fnOptions": undefined,
              "fnScopeId": undefined,
              "isAsyncPlaceholder": false,
              "isCloned": false,
              "isComment": false,
              "isOnce": false,
              "isRootInsert": false,
              "isStatic": false,
              "key": undefined,
              "ns": undefined,
              "parent": undefined,
              "raw": false,
              "tag": undefined,
              "text": "",
            },
          ],
          "componentInstance": undefined,
          "componentOptions": undefined,
          "context": [Circular],
          "data": undefined,
          "elm": <h1>
            
          </h1>,
          "fnContext": undefined,
          "fnOptions": undefined,
          "fnScopeId": undefined,
          "isAsyncPlaceholder": false,
          "isCloned": false,
          "isComment": false,
          "isOnce": false,
          "isRootInsert": false,
          "isStatic": false,
          "key": undefined,
          "ns": undefined,
          "parent": undefined,
          "raw": false,
          "tag": "h1",
          "text": undefined,
        },
      ],
      "componentInstance": undefined,
      "componentOptions": undefined,
      "context": [Circular],
      "data": Object {
        "staticClass": "hello",
      },
      "elm": <div
        class="hello"
      >
        <h1>
          
        </h1>
      </div>,
      "fnContext": undefined,
      "fnOptions": undefined,
      "fnScopeId": undefined,
      "isAsyncPlaceholder": false,
      "isCloned": false,
      "isComment": false,
      "isOnce": false,
      "isRootInsert": true,
      "isStatic": false,
      "key": undefined,
      "ns": undefined,
      "parent": undefined,
      "raw": false,
      "tag": "div",
      "text": undefined,
    },
    "_watcher": Watcher {
      "active": true,
      "cb": [Function],
      "deep": false,
      "depIds": Set {
        6,
      },
      "deps": Array [
        Dep {
          "id": 6,
          "subs": Array [
            [Circular],
          ],
        },
      ],
      "dirty": false,
      "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
      "getter": [Function],
      "id": 2,
      "lazy": false,
      "newDepIds": Set {},
      "newDeps": Array [],
      "sync": false,
      "user": false,
      "value": undefined,
      "vm": [Circular],
    },
    "_watchers": Array [
      Watcher {
        "active": true,
        "cb": [Function],
        "deep": false,
        "depIds": Set {
          6,
        },
        "deps": Array [
          Dep {
            "id": 6,
            "subs": Array [
              [Circular],
            ],
          },
        ],
        "dirty": false,
        "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
        "getter": [Function],
        "id": 2,
        "lazy": false,
        "newDepIds": Set {},
        "newDeps": Array [],
        "sync": false,
        "user": false,
        "value": undefined,
        "vm": [Circular],
      },
    ],
  },
  "vnode": VNode {
    "asyncFactory": undefined,
    "asyncMeta": undefined,
    "children": Array [
      VNode {
        "asyncFactory": undefined,
        "asyncMeta": undefined,
        "children": Array [
          VNode {
            "asyncFactory": undefined,
            "asyncMeta": undefined,
            "children": undefined,
            "componentInstance": undefined,
            "componentOptions": undefined,
            "context": undefined,
            "data": undefined,
            "elm": ,
            "fnContext": undefined,
            "fnOptions": undefined,
            "fnScopeId": undefined,
            "isAsyncPlaceholder": false,
            "isCloned": false,
            "isComment": false,
            "isOnce": false,
            "isRootInsert": false,
            "isStatic": false,
            "key": undefined,
            "ns": undefined,
            "parent": undefined,
            "raw": false,
            "tag": undefined,
            "text": "",
          },
        ],
        "componentInstance": undefined,
        "componentOptions": undefined,
        "context": VueComponent {
          "$_mountingOptionsSlots": undefined,
          "$_originalSlots": Object {},
          "$attrs": Object {},
          "$children": Array [],
          "$createElement": [Function],
          "$el": <div
            class="hello"
          >
            <h1>
              
            </h1>
          </div>,
          "$emit": [Function],
          "$listeners": Object {},
          "$options": Object {
            "_Ctor": Object {
              "0": [Function],
            },
            "_base": [Function],
            "_parentElm": null,
            "_propKeys": Array [
              "message",
            ],
            "_refElm": null,
            "beforeCreate": Array [
              [Function],
            ],
            "components": Object {
              "transition": Object {
                "render": [Function],
              },
              "transition-group": Object {
                "render": [Function],
              },
            },
            "data": [Function],
            "directives": Object {},
            "filters": Object {},
            "mixins": Array [
              Object {
                "data": [Function],
              },
            ],
            "name": "HelloWithProps",
            "props": Object {
              "message": Object {
                "type": [Function],
              },
            },
            "render": [Function],
            "staticRenderFns": Array [],
          },
          "$parent": undefined,
          "$refs": Object {},
          "$root": [Circular],
          "$scopedSlots": Object {},
          "$slots": Object {},
          "$vnode": undefined,
          "__emitted": Object {},
          "__emittedByOrder": Array [],
          "_c": [Function],
          "_data": Object {},
          "_directInactive": false,
          "_events": Object {},
          "_hasHookEvent": false,
          "_inactive": null,
          "_isBeingDestroyed": false,
          "_isDestroyed": false,
          "_isMounted": true,
          "_isVue": true,
          "_props": Object {
            "message": undefined,
          },
          "_renderProxy": VueComponent {
            "$_mountingOptionsSlots": undefined,
            "$_originalSlots": Object {},
            "$attrs": Object {},
            "$children": Array [],
            "$createElement": [Function],
            "$el": <div
              class="hello"
            >
              <h1>
                
              </h1>
            </div>,
            "$emit": [Function],
            "$listeners": Object {},
            "$options": Object {
              "_Ctor": Object {
                "0": [Function],
              },
              "_base": [Function],
              "_parentElm": null,
              "_propKeys": Array [
                "message",
              ],
              "_refElm": null,
              "beforeCreate": Array [
                [Function],
              ],
              "components": Object {
                "transition": Object {
                  "render": [Function],
                },
                "transition-group": Object {
                  "render": [Function],
                },
              },
              "data": [Function],
              "directives": Object {},
              "filters": Object {},
              "mixins": Array [
                Object {
                  "data": [Function],
                },
              ],
              "name": "HelloWithProps",
              "props": Object {
                "message": Object {
                  "type": [Function],
                },
              },
              "render": [Function],
              "staticRenderFns": Array [],
            },
            "$parent": undefined,
            "$refs": Object {},
            "$root": [Circular],
            "$scopedSlots": Object {},
            "$slots": Object {},
            "$vnode": undefined,
            "__emitted": Object {},
            "__emittedByOrder": Array [],
            "_c": [Function],
            "_data": Object {},
            "_directInactive": false,
            "_events": Object {},
            "_hasHookEvent": false,
            "_inactive": null,
            "_isBeingDestroyed": false,
            "_isDestroyed": false,
            "_isMounted": true,
            "_isVue": true,
            "_props": Object {
              "message": undefined,
            },
            "_renderProxy": [Circular],
            "_self": [Circular],
            "_staticTrees": null,
            "_uid": 1,
            "_vnode": [Circular],
            "_watcher": Watcher {
              "active": true,
              "cb": [Function],
              "deep": false,
              "depIds": Set {
                6,
              },
              "deps": Array [
                Dep {
                  "id": 6,
                  "subs": Array [
                    [Circular],
                  ],
                },
              ],
              "dirty": false,
              "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
              "getter": [Function],
              "id": 2,
              "lazy": false,
              "newDepIds": Set {},
              "newDeps": Array [],
              "sync": false,
              "user": false,
              "value": undefined,
              "vm": [Circular],
            },
            "_watchers": Array [
              Watcher {
                "active": true,
                "cb": [Function],
                "deep": false,
                "depIds": Set {
                  6,
                },
                "deps": Array [
                  Dep {
                    "id": 6,
                    "subs": Array [
                      [Circular],
                    ],
                  },
                ],
                "dirty": false,
                "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
                "getter": [Function],
                "id": 2,
                "lazy": false,
                "newDepIds": Set {},
                "newDeps": Array [],
                "sync": false,
                "user": false,
                "value": undefined,
                "vm": [Circular],
              },
            ],
          },
          "_self": [Circular],
          "_staticTrees": null,
          "_uid": 1,
          "_vnode": [Circular],
          "_watcher": Watcher {
            "active": true,
            "cb": [Function],
            "deep": false,
            "depIds": Set {
              6,
            },
            "deps": Array [
              Dep {
                "id": 6,
                "subs": Array [
                  [Circular],
                ],
              },
            ],
            "dirty": false,
            "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
            "getter": [Function],
            "id": 2,
            "lazy": false,
            "newDepIds": Set {},
            "newDeps": Array [],
            "sync": false,
            "user": false,
            "value": undefined,
            "vm": [Circular],
          },
          "_watchers": Array [
            Watcher {
              "active": true,
              "cb": [Function],
              "deep": false,
              "depIds": Set {
                6,
              },
              "deps": Array [
                Dep {
                  "id": 6,
                  "subs": Array [
                    [Circular],
                  ],
                },
              ],
              "dirty": false,
              "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
              "getter": [Function],
              "id": 2,
              "lazy": false,
              "newDepIds": Set {},
              "newDeps": Array [],
              "sync": false,
              "user": false,
              "value": undefined,
              "vm": [Circular],
            },
          ],
        },
        "data": undefined,
        "elm": <h1>
          
        </h1>,
        "fnContext": undefined,
        "fnOptions": undefined,
        "fnScopeId": undefined,
        "isAsyncPlaceholder": false,
        "isCloned": false,
        "isComment": false,
        "isOnce": false,
        "isRootInsert": false,
        "isStatic": false,
        "key": undefined,
        "ns": undefined,
        "parent": undefined,
        "raw": false,
        "tag": "h1",
        "text": undefined,
      },
    ],
    "componentInstance": undefined,
    "componentOptions": undefined,
    "context": VueComponent {
      "$_mountingOptionsSlots": undefined,
      "$_originalSlots": Object {},
      "$attrs": Object {},
      "$children": Array [],
      "$createElement": [Function],
      "$el": <div
        class="hello"
      >
        <h1>
          
        </h1>
      </div>,
      "$emit": [Function],
      "$listeners": Object {},
      "$options": Object {
        "_Ctor": Object {
          "0": [Function],
        },
        "_base": [Function],
        "_parentElm": null,
        "_propKeys": Array [
          "message",
        ],
        "_refElm": null,
        "beforeCreate": Array [
          [Function],
        ],
        "components": Object {
          "transition": Object {
            "render": [Function],
          },
          "transition-group": Object {
            "render": [Function],
          },
        },
        "data": [Function],
        "directives": Object {},
        "filters": Object {},
        "mixins": Array [
          Object {
            "data": [Function],
          },
        ],
        "name": "HelloWithProps",
        "props": Object {
          "message": Object {
            "type": [Function],
          },
        },
        "render": [Function],
        "staticRenderFns": Array [],
      },
      "$parent": undefined,
      "$refs": Object {},
      "$root": [Circular],
      "$scopedSlots": Object {},
      "$slots": Object {},
      "$vnode": undefined,
      "__emitted": Object {},
      "__emittedByOrder": Array [],
      "_c": [Function],
      "_data": Object {},
      "_directInactive": false,
      "_events": Object {},
      "_hasHookEvent": false,
      "_inactive": null,
      "_isBeingDestroyed": false,
      "_isDestroyed": false,
      "_isMounted": true,
      "_isVue": true,
      "_props": Object {
        "message": undefined,
      },
      "_renderProxy": VueComponent {
        "$_mountingOptionsSlots": undefined,
        "$_originalSlots": Object {},
        "$attrs": Object {},
        "$children": Array [],
        "$createElement": [Function],
        "$el": <div
          class="hello"
        >
          <h1>
            
          </h1>
        </div>,
        "$emit": [Function],
        "$listeners": Object {},
        "$options": Object {
          "_Ctor": Object {
            "0": [Function],
          },
          "_base": [Function],
          "_parentElm": null,
          "_propKeys": Array [
            "message",
          ],
          "_refElm": null,
          "beforeCreate": Array [
            [Function],
          ],
          "components": Object {
            "transition": Object {
              "render": [Function],
            },
            "transition-group": Object {
              "render": [Function],
            },
          },
          "data": [Function],
          "directives": Object {},
          "filters": Object {},
          "mixins": Array [
            Object {
              "data": [Function],
            },
          ],
          "name": "HelloWithProps",
          "props": Object {
            "message": Object {
              "type": [Function],
            },
          },
          "render": [Function],
          "staticRenderFns": Array [],
        },
        "$parent": undefined,
        "$refs": Object {},
        "$root": [Circular],
        "$scopedSlots": Object {},
        "$slots": Object {},
        "$vnode": undefined,
        "__emitted": Object {},
        "__emittedByOrder": Array [],
        "_c": [Function],
        "_data": Object {},
        "_directInactive": false,
        "_events": Object {},
        "_hasHookEvent": false,
        "_inactive": null,
        "_isBeingDestroyed": false,
        "_isDestroyed": false,
        "_isMounted": true,
        "_isVue": true,
        "_props": Object {
          "message": undefined,
        },
        "_renderProxy": [Circular],
        "_self": [Circular],
        "_staticTrees": null,
        "_uid": 1,
        "_vnode": [Circular],
        "_watcher": Watcher {
          "active": true,
          "cb": [Function],
          "deep": false,
          "depIds": Set {
            6,
          },
          "deps": Array [
            Dep {
              "id": 6,
              "subs": Array [
                [Circular],
              ],
            },
          ],
          "dirty": false,
          "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
          "getter": [Function],
          "id": 2,
          "lazy": false,
          "newDepIds": Set {},
          "newDeps": Array [],
          "sync": false,
          "user": false,
          "value": undefined,
          "vm": [Circular],
        },
        "_watchers": Array [
          Watcher {
            "active": true,
            "cb": [Function],
            "deep": false,
            "depIds": Set {
              6,
            },
            "deps": Array [
              Dep {
                "id": 6,
                "subs": Array [
                  [Circular],
                ],
              },
            ],
            "dirty": false,
            "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
            "getter": [Function],
            "id": 2,
            "lazy": false,
            "newDepIds": Set {},
            "newDeps": Array [],
            "sync": false,
            "user": false,
            "value": undefined,
            "vm": [Circular],
          },
        ],
      },
      "_self": [Circular],
      "_staticTrees": null,
      "_uid": 1,
      "_vnode": [Circular],
      "_watcher": Watcher {
        "active": true,
        "cb": [Function],
        "deep": false,
        "depIds": Set {
          6,
        },
        "deps": Array [
          Dep {
            "id": 6,
            "subs": Array [
              [Circular],
            ],
          },
        ],
        "dirty": false,
        "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
        "getter": [Function],
        "id": 2,
        "lazy": false,
        "newDepIds": Set {},
        "newDeps": Array [],
        "sync": false,
        "user": false,
        "value": undefined,
        "vm": [Circular],
      },
      "_watchers": Array [
        Watcher {
          "active": true,
          "cb": [Function],
          "deep": false,
          "depIds": Set {
            6,
          },
          "deps": Array [
            Dep {
              "id": 6,
              "subs": Array [
                [Circular],
              ],
            },
          ],
          "dirty": false,
          "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
          "getter": [Function],
          "id": 2,
          "lazy": false,
          "newDepIds": Set {},
          "newDeps": Array [],
          "sync": false,
          "user": false,
          "value": undefined,
          "vm": [Circular],
        },
      ],
    },
    "data": Object {
      "staticClass": "hello",
    },
    "elm": <div
      class="hello"
    >
      <h1>
        
      </h1>
    </div>,
    "fnContext": undefined,
    "fnOptions": undefined,
    "fnScopeId": undefined,
    "isAsyncPlaceholder": false,
    "isCloned": false,
    "isComment": false,
    "isOnce": false,
    "isRootInsert": true,
    "isStatic": false,
    "key": undefined,
    "ns": undefined,
    "parent": undefined,
    "raw": false,
    "tag": "div",
    "text": undefined,
  },
}
`;
